streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamSessionWindowAggregateProcessorTest.java

```diff
@@ -590,41 +560,33 @@ private void shouldLogAndMeterWhenSkippingLateRecordWithNonzeroGrace(final Strin
             );
         }
 
-        assertThat(metrics.metrics().get(dropTotal).metricValue(), is(1.0));
+        assertThat(context.metrics().metrics().get(dropTotal).metricValue(), is(1.0));
         assertThat(
-            (Double) metrics.metrics().get(dropRate).metricValue(),
+            (Double) context.metrics().metrics().get(dropRate).metricValue(),
             greaterThan(0.0));
         assertThat(
             appender.getMessages(),
             hasItem("Skipping record for expired window. key=[Late1] topic=[topic] partition=[-3] offset=[-2] timestamp=[0] window=[0,0] expiration=[1] streamTime=[2]"));
     }
 
-    private InternalMockProcessorContext createInternalMockProcessorContext(final String builtInMetricsVersion) {
-        final StreamsMetricsImpl streamsMetrics = new StreamsMetricsImpl(metrics, "test", builtInMetricsVersion);
-        final InternalMockProcessorContext context = new InternalMockProcessorContext(
-            TestUtils.tempDirectory(),
-            Serdes.String(),
-            Serdes.String(),
-            streamsMetrics,
-            new StreamsConfig(StreamsTestUtils.getStreamsConfig()),
-            MockRecordCollector::new,
-            new ThreadCache(new LogContext("testCache "), 100000, streamsMetrics)
-        ) {
-            @Override
-            public <K, V> void forward(final K key, final V value, final To to) {
-                toInternal.update(to);
-                results.add(new KeyValueTimestamp<>(key, value, toInternal.timestamp()));
-            }
-        };
-        TaskMetrics.droppedRecordsSensorOrSkippedRecordsSensor(threadId, context.taskId().toString(), streamsMetrics);
+    private static InternalProcessorContext createInternalProcessorContext(final String builtInMetricsVersion) {
+        final Properties streamsConfig = StreamsTestUtils.getStreamsConfig();
+        streamsConfig.setProperty(StreamsConfig.BUILT_IN_METRICS_VERSION_CONFIG, builtInMetricsVersion);
+        final MockInternalProcessorContext context = new MockInternalProcessorContext(streamsConfig);
+
         final StoreBuilder<SessionStore<String, Long>> storeBuilder =
             Stores.sessionStoreBuilder(
                 Stores.persistentSessionStore(STORE_NAME, ofMillis(GAP_MS * 3)),
                 Serdes.String(),
                 Serdes.Long())
                 .withLoggingDisabled();
-        final SessionStore sessionStore = storeBuilder.build();
+        final SessionStore<String, Long> sessionStore = storeBuilder.build();
         sessionStore.init(context, sessionStore);
         return context;
     }
+
+    private List<KeyValueTimestamp<Object, Object>> results() {
+        return context.forwarded().stream().map(cf -> new KeyValueTimestamp<>(cf.keyValue().key, cf.keyValue().value, cf.timestamp())).collect(Collectors.toList());
+    }
```

Comment:
prop: Here you could fix the generics, but that would mean a bit of work in `MockProcessorContext`, I guess. This is not required for this PR, but if you want you are very welcome to fix this in new PR. I opened ticket [KAFKA-9738](https://issues.apache.org/jira/browse/KAFKA-9738) to document this issue.